src/plugins/home/server/services/sample_data/routes/uninstall.ts

```diff
@@ -81,17 +85,11 @@ export function createUninstallRoute(
       }
 
       let savedObjectsList = sampleDataset.savedObjects;
-      if (dataSourceId) {
-        savedObjectsList = sampleDataset.getDataSourceIntegratedSavedObjects(
-          savedObjectsList,
-          dataSourceId
-        );
-      }
       if (workspaceId) {
-        savedObjectsList = sampleDataset.getWorkspaceIntegratedSavedObjects(
-          savedObjectsList,
-          workspaceId
-        );
+        savedObjectsList = getWorkspaceIntegratedSavedObjects(savedObjectsList, workspaceId);
+      }
+      if (dataSourceId) {
+        savedObjectsList = getDataSourceIntegratedSavedObjects(savedObjectsList, dataSourceId);
```

Comment:
I think we should keep the same id generate order with install. We prepend the data source id first in install file.

Reply:
You're right, I missed that, I was trying to align with the convention of `getDashboardWithPrefix` and `getDataSourceIntegratedDefaultIndex` which id is like `<data_source_id>_<workspace_id>_<object_id>`, do you have concern if we change the order like this at both `install` and `uninstall`?

Reply:
Yeah. I think we can keep the same rule with `getDashboardWithPrefix` and `getDataSourceIntegratedDefaultIndex`